File: Prb_1179.java
/**
 * Prb_1179
 */
import java.util.Scanner;
public class Prb_1179 {

    public static void main(String[] args){
        Scanner s = new Scanner(System.in);
        int count = 0, even[]= new int[5], odd[] = new int[5], number = 0,ec = 0, oc = 0 ;
        while(count < 15){
            number = s.nextInt();
            if(number % 2 == 0){
                even[ec] = number;
                ec++;
                if(ec % 5 == 0){
                    int i = 0;
                    while(i < 5){
                        System.out.println("par["+i+"] = "+even[i]);
                        ++i;
                    }
                    ec = 0; 
                }
            }else{
                odd[oc] = number;
                oc++;
                if(oc % 5 == 0){
                    int i = 0;
                    while(i < 5){
                        System.out.println("impar["+i+"] = "+odd[i]);
                        ++i;
                    }
                    oc = 0;
                }        
            }
            count++;
        }
        int remainingOdd = (oc % 5);
        int remainingEven = (ec % 5);
        if(remainingOdd > 0){
            int i = 0;
            while(i < remainingOdd){
                System.out.println("impar["+i+"] = "+odd[i]);
                ++i;
            }
        }
        if(remainingEven > 0){
            int i = 0;
            while(i < remainingEven){
                System.out.println("par["+i+"] = "+even[i]);
                ++i;
            }
        }
    }
}

File: Prb_1179.cpp
//Not solved
#include <iostream>
using namespace std;
int main()
{
    int a1[5],a2[5],count {0},num,c1 {0},c2 {0};
    while (count < 15)
    {
        cin>>num;
        if(num % 2 == 0 & c1 < 5){
            a1[c1] = num;
            ++c1;
            if(c1 == 4){
                for (int i : a1)
                {
                    cout<<"par["<<i<<"] = "<< a1 <<endl;
                }
                c1 = 0;  
            }
        }else{
            a2[c2] = num;
            ++c2;
            if(c2 == 4){
                for (int i : a2)
                {
                    cout<<"par["<<i<<"] = "<< a2 <<endl;
                }
                c2 = 0;
            }
        }
    }
    
    

    return 0;
}